### PR TITLE
Fix set optimizer attributes

### LINKED ISSUE #68: Document setting attributes using `set_optimizer_attributes` or `optimizer_with_attributes`
The JuMP documentation says that a user can use `optimizer_with_attributes` which calls `set_optimizer_attribute`.

https://github.com/JuliaOpt/JuMP.jl/blob/13ad5db156210f805699e8a1004610d60bc3be3b/src/JuMP.jl#L56-L89

~Current Xpress.jl does not implement `set_optimizer_attribute`.~

It seems that we don't need to implement `set_optimizer_attribute` but currently the attributes don't propogate to the actual model? Maybe I'm missing something or just don't know how to do this. I'm opening this issue so that I can keep track and at the very least submit a PR for documentation.

### PR DESCRIPTION
**Pull request by kdheepak**
All setting parameters from JuMP and propogate the values set to the XpressProblem actually being used for solving.

Fixes #68

### CODE REVIEW

File: src/MOI/MOI_wrapper.jl [hunk 1]
```diff
@@ -51,6 +51,12 @@ const SCALAR_SETS = Union{
     MOI.Interval{Float64},
 }
 
+const XPRS_ATTRIBUTES = Dict{String, Any}(
+    replace(string(name), "XPRS_"=>"") => getfield(Lib, name)
+        for name in names(Lib; all = true)
+        if startswith(string(name), "XPRS_") && all(isuppercase(c) for c in string(name) if c != '_')
+)
+
```
Comment (joaquimg) [inline src/MOI/MOI_wrapper.jl]: I had already implmented similar stuff in:
```
    const parameter_name_to_index = Dict{String, Int}(
        string(name) => getfield(Lib, name)
        for name in names(Lib; all = true)
        if startswith(string(name), "XPRS_") && typeof(getfield(Lib, name)) == Int
    )
```

**Comment by joaquimg** [comment]
bump

File: src/MOI/MOI_wrapper.jl (hunk 1 above)
Comment (kdheepak) [reply src/MOI/MOI_wrapper.jl]: Fixed in c2f5ffbeda74470da463301f002da1c95704a628. 

I chose to remove `parameter_name_to_index` and use `XPRS_ATTRIBUTES` everywhere.

1) using `typeof(getfield(Lib, name)) == Int` does not get attributes that are not integers e.g.

```
const XPRS_OBJ_MINIMIZE = 1.0
const XPRS_OBJ_MAXIMIZE = -1.0
...
const XPRS_ISUSERSOLUTION = 0x01
const XPRS_ISREPROCESSEDUSERSOLUTION = 0x02
```

2) I think we don't want to require that the user type `XPRS_$attribute` every time. I don't feel strongly about this though.

Thoughts?

**Comment by kdheepak** [comment]
I rebased and fixed all merge conflicts on top of latest `master`.

File: src/MOI/MOI_wrapper.jl (hunk 1 above)
Comment (joaquimg) [reply src/MOI/MOI_wrapper.jl]: agree with 1 and 2

**Comment by kdheepak** [comment]
Good idea! I'll do it right now.

**Comment by joaquimg** [comment]
Oops, I meant a example in the readme!
Adding JuMP as a dependency might be too much

**Comment by kdheepak** [comment]
Yes, I was going to update README and add a test.

**Comment by joaquimg** [comment]
Perfect! Lets not test JuMP directly. You can test the attribute passing in MOI.
Tests depending on JuMP is bad, because MOI move faster.

**Comment by kdheepak** [comment]
See 97a631d for a single test. We can add more `MOI.RawParameter` tests later.

**Comment by joaquimg** [comment]
Looks good to me!

**Pull request merged**